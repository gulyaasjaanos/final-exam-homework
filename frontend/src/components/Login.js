import React from 'react';
import { useEffect, useState } from 'react';
import sessionService from '../services/sessionService';
import { Redirect } from 'react-router-dom';
import { consoleLogAction, setNameAction } from '../actions';
import { connect } from 'react-redux';

function Login({console, consoleLog, sessionUser, setName}) {
  
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [loggedIn, setLoggedIn] = useState(false)

  useEffect( () => {  

    sessionService.session()
      .then( session => {
        if (session) setLoggedIn(true)
      });

  }, []);

  
  const login = async () => {
    const logged = await sessionService.login({ username, password });
    if (logged.token) {
        localStorage.setItem('token',  logged.token);
        localStorage.setItem('username',  username);
        setLoggedIn(true);
        consoleLog('');
        setName(username);
    } else {
      consoleLog(logged.error);
    }
  };

  const changeUsername = (e) => {
    setUsername(e.target.value);
  };

  const changePassword = (e) => {
    setPassword(e.target.value);
  }; 

  if (loggedIn) {
    return (<Redirect push to={'/items'} />);
  } else {
    return (
      <nav>
        <header>Hi Stranger</header>
        <input type="text" value={username} onChange={(event) => changeUsername(event)}></input>
        <input type="password" value={password} onChange={(event) => changePassword(event)}></input>
        <button onClick={login}>LOGIN</button>
      </nav>
    );
  }

}

const mapStateToProps = state => (
  {
    console: state.console,
    sessionUser: state.session
  }
);


const mapDispatchToProps = dispatch => (
  {
    consoleLog: (message) => dispatch( consoleLogAction(message) ),
    setName: (username) => dispatch( setNameAction(username) )
  }
);
 

export default connect(mapStateToProps,mapDispatchToProps)( Login );
