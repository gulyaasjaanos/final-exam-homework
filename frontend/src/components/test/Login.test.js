import React from 'react';
import { Provider } from 'react-redux';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import configureStore from 'redux-mock-store';

import Login from '../Login';

let container;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('matches snapeshot', async () => {
  const mockStore = configureStore([]);
  const mockedStore = mockStore({
  });
  mockedStore.dispatch = jest.fn();
  await act(async () => {
    render(
      <Provider store={mockedStore}>
        <Login />
      </Provider>,
      container
    );
  });

  expect(container.innerHTML).toMatchSnapshot();
});

