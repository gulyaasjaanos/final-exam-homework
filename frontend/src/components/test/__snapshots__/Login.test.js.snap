// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapeshot 1`] = `"<nav><header>Hi Stranger</header><input type=\\"text\\" placeholder=\\"username\\" value=\\"\\"><input type=\\"password\\" placeholder=\\"password\\" value=\\"\\"><button>LOGIN</button></nav>"`;
